pkg/cmd/admin/router/router.go

```diff
@@ -505,154 +546,183 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
+				ObjectMeta: kapi.ObjectMeta{Name: fmt.Sprintf("router-%s-role", cfg.Name)},
+				Subjects: []kapi.ObjectReference{
+					{
+						Kind:      "ServiceAccount",
+						Name:      cfg.ServiceAccount,
+						Namespace: namespace,
+					},
+				},
+				RoleRef: kapi.ObjectReference{
+					Kind: "ClusterRole",
+					Name: "system:router",
+				},
+			},
+		)
+	}
+	objects = append(objects, &deployapi.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:   name,
+			Labels: label,
+		},
+		Spec: deployapi.DeploymentConfigSpec{
+			Strategy: deployapi.DeploymentStrategy{
+				Type:          deployapi.DeploymentStrategyTypeRolling,
+				RollingParams: &deployapi.RollingDeploymentStrategyParams{UpdatePercent: &updatePercent},
+			},
+			Replicas: cfg.Replicas,
+			Selector: label,
+			Triggers: []deployapi.DeploymentTriggerPolicy{
+				{Type: deployapi.DeploymentTriggerOnConfigChange},
+			},
+			Template: &kapi.PodTemplateSpec{
+				ObjectMeta: kapi.ObjectMeta{Labels: label},
+				Spec: kapi.PodSpec{
+					SecurityContext: &kapi.PodSecurityContext{
+						HostNetwork: cfg.HostNetwork,
+					},
+					ServiceAccountName: cfg.ServiceAccount,
+					NodeSelector:       nodeSelector,
+					Containers:         containers,
+					Volumes:            volumes,
+				},
+			},
+		},
+	})
+
+	objects = app.AddServices(objects, false)
+	// set the service port to the provided hostport value
+	for i := range objects {
+		switch t := objects[i].(type) {
+		case *kapi.Service:
+			for j, servicePort := range t.Spec.Ports {
+				for _, targetPort := range ports {
+					if targetPort.ContainerPort == servicePort.Port && targetPort.HostPort != 0 {
+						t.Spec.Ports[j].Port = targetPort.HostPort
+					}
+				}
 			}
-			return nil
 		}
+	}
+	// TODO: label all created objects with the same label - router=<name>
+	list := &kapi.List{Items: objects}
 
-		mapper, typer := f.Factory.Object()
-		bulk := configcmd.Bulk{
-			Mapper:            mapper,
-			Typer:             typer,
-			RESTClientFactory: f.Factory.ClientForMapping,
-
-			After: configcmd.NewPrintNameOrErrorAfter(mapper, cmdutil.GetFlagString(cmd, "output") == "name", "created", out, cmd.Out()),
+	if output {
+		list.Items, err = ocmdutil.ConvertItemsForDisplayFromDefaultCommand(cmd, list.Items)
+		if err != nil {
+			return err
 		}
-		if errs := bulk.Create(list, namespace); len(errs) != 0 {
-			return errExit
+
+		if err := f.PrintObject(cmd, list, out); err != nil {
+			return fmt.Errorf("Unable to print object: %v", err)
 		}
 		return nil
 	}
 
-	fmt.Fprintf(out, "Router %q service exists\n", name)
+	mapper, typer := f.Factory.Object()
+	bulk := configcmd.Bulk{
+		Mapper:            mapper,
+		Typer:             typer,
+		RESTClientFactory: f.Factory.ClientForMapping,
+
+		After: configcmd.NewPrintNameOrErrorAfter(mapper, cmdutil.GetFlagString(cmd, "output") == "name", "created", out, cmd.Out()),
+	}
+	if errs := bulk.Create(list, namespace); len(errs) != 0 {
+		return errExit
```

Comment:
we print in order of user input here, vs returning user successes then
failures.  I like interspersed execution better.

On Wed, Feb 17, 2016 at 12:29 PM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/cmd/admin/router/router.go
> <https://github.com/openshift/origin/pull/7251#discussion_r53199657>:
>
> >  		}
> >  		return nil
> >  	}
> >
> > -	fmt.Fprintf(out, "Router %q service exists\n", name)
> > +	mapper, typer := f.Factory.Object()
> > +	bulk := configcmd.Bulk{
> > +		Mapper:            mapper,
> > +		Typer:             typer,
> > +		RESTClientFactory: f.Factory.ClientForMapping,
> > +
> > +		After: configcmd.NewPrintNameOrErrorAfter(mapper, cmdutil.GetFlagString(cmd, "output") == "name", "created", out, cmd.Out()),
> > +	}
> > +	if errs := bulk.Create(list, namespace); len(errs) != 0 {
> > +		return errExit
>
> Is errExit more informative than errs aggregated?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7251/files#r53199657>.
>